<launch>
  <arg name="host_ip"/>
  <arg name="localization_method"/>
  <arg name="app_name"/>
  <arg name="DEBUG"/>
  <arg name="update_coverage_freq"/>
  <arg name="coverage_threshold"/>
  <arg name="mav_name" default="airsim_qc" />
  <arg name="waypoint_x" default="0" />
  <arg name="waypoint_y" default="0" />
  <arg name="waypoint_z" default="1" />
  <arg name="subscript" default="" />
  <arg name="enable_logging" default="false" />
  <arg name="enable_ground_truth" default="true" />
  <arg name="log_file" default="$(arg mav_name)" />
  <arg name="param_file" default="$(find nbvplanner)/resource/mapping.yaml" />
  <arg name="tf_frame" default="navigation" />
  <arg name="pcl_topic" default="/pointcloudOut" />
  <arg name="stl_file_path" default="" />
  <arg name="mesh_resolution" default="1.0" />
  <arg name="resolution" default="0.4" />
  <arg name="visualize_max_z" default="999" />
  <arg name="sensor_max_range" default="100.0" />
  <arg name="peer_vehicle_pose_topic_1" default="peer_pose_1" />
  <arg name="peer_vehicle_pose_topic_2" default="peer_pose_2" />
  <arg name="peer_vehicle_pose_topic_3" default="peer_pose_3" />
  
  <param name="ip_addr" type="string" value="$(arg host_ip)" />
  <param name="localization_method" type="string" value="$(arg localization_method)" />
  <param name="coverage_threshold" value="$(arg coverage_threshold)"/> 

  <!---nodes /-->
  <!--node pkg="tf" type="static_transform_publisher" name="tf_1$(arg subscript)" args="0 0 0 0 0 0 $(arg mav_name)$(arg subscript)/vi_sensor/base_link world$(arg subscript) 10" /-->
  <!--node pkg="tf" type="static_transform_publisher" name="tf_2$(arg subscript)" args="0.015 0.055 0.0065 -1.57 0.1 -1.57 world$(arg subscript) $(arg mav_name)$(arg
  subscript)/vi_sensor/camera_depth_optical_center_link 1" /-->
  <!--node pkg="tf" type="static_transform_publisher" name="tf_3$(arg subscript)" args="0.015 0.055 0.0065 -1.57 0.1 -1.57 world$(arg subscript) $(arg mav_name)$(arg subscript)/vi_sensor/camera_left_link
  1" /-->
  <!--node pkg="tf" type="static_transform_publisher" name="tf_4$(arg subscript)" args="0.015 -0.055 0.0065 -1.57 0.1 -1.57 world$(arg subscript) $(arg mav_name)$(arg
  subscript)/vi_sensor/camera_right_link 1" /-->
  
  <comment launch-prefix="xterm -e gdb --args" />
  <comment respawn="false" launch-prefix="gdb -ex run --args" />
  
  <!-- ns is shortened for namespace that will be prepended to all the topics --> 
  <group ns="$(arg mav_name)$(arg subscript)">
      <param name="wp_x" type="double" value="$(arg waypoint_x)" />
      <param name="wp_y" type="double" value="$(arg waypoint_y)" />
      <param name="wp_z" type="double" value="$(arg waypoint_z)" />
      <!-- arg name="mav_name" value="$(arg mav_name)$(arg subscript)" /-->
      <!-- arg name="enable_logging" value="$(arg enable_logging)" /-->
      <!-- arg name="enable_ground_truth" value="$(arg enable_ground_truth)" /-->
      <!-- arg name="log_file" value="$(arg log_file)"/-->
      <arg name="x" value="$(arg waypoint_x)"/>
      <arg name="y" value="$(arg waypoint_y)"/>
      <arg name="z" value="0.1"/>
      <!--arg name="tf_prefix" default="$(arg mav_name)$(arg subscript)"/-->
     
      <node name="nbvPlanner" pkg="nbvplanner" type="nbvPlanner" output="screen">  
          <!--param name="tf_frame" type="string" value="$(arg tf_frame)" /-->
          <param name="DEBUG" value="$(arg DEBUG)"/> 
          <param name="update_coverage_freq" value="$(arg update_coverage_freq)"/> 
          <param name="tf_frame" type="string" value="world" />
          <param name="resolution" type="double" value="$(arg resolution)" />
          <param name="visualize_max_z" type="double" value="$(arg visualize_max_z)" />
          <param name="sensor_max_range" type="double" value="$(arg sensor_max_range)" />
          <param name="stl_file_path" type="string" value="$(arg stl_file_path)" />
          <param name="mesh_resolution" type="double" value="$(arg mesh_resolution)" />
          <param name="map_publish_frequency" type="double" value="20.0" />
          <!--param name="ip_addr" type="string" value="/$(arg mav_name)$(arg subscript)$(arg host_ip)" /-->
          <remap from="cam0/camera_info" to="/$(arg mav_name)$(arg subscript)/vi_sensor/left/camera_info"/>
          <remap from="cam1/camera_info" to="/$(arg mav_name)$(arg subscript)/vi_sensor/right/camera_info"/>
          <remap from="pointcloud_throttled" to="$(arg pcl_topic)"/>
          <remap from="pose" to="/pose_topic"/>
          <remap from="nbvplanner" to="/$(arg mav_name)$(arg subscript)/nbvplanner"/>
          <remap from="peer_pose_1" to="$(arg peer_vehicle_pose_topic_1)"/>
          <remap from="peer_pose_2" to="$(arg peer_vehicle_pose_topic_2)"/>
          <remap from="peer_pose_3" to="$(arg peer_vehicle_pose_topic_3)"/>
          <param name="tf_frame" type="string" value="world" />
          <param name="robot_frame" type="string" value="$(arg localization_method)"/>
          <remap from="disparity" to="/Airsim/disparity"/>
          <remap from="pointcloud" to="/points"/>
          <remap from="cam0/camera_info" to="/Airsim/left/camera_info"/>
          <remap from="cam1/camera_info" to="/Airsim/right/camera_info"/>
          <rosparam command="load" file="$(arg param_file)" />
      </node>
        
      <!--node name="mapping" pkg="mapping_and_sar" type="mapping" output="screen"-->
      <node name="$(arg app_name)" pkg="mapping_and_sar" type="$(arg app_name)" output="screen">
        <!--param name="localization_method" type="string" value="$(arg localization_method)"/-->
        <rosparam command="load" file="$(arg param_file)" />
      </node>
  </group>
</launch>
